drivers/123_open/types.go

```diff
@@ -0,0 +1,197 @@
+package _123_open
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/OpenListTeam/OpenList/internal/model"
+	"github.com/OpenListTeam/OpenList/pkg/utils"
+)
+
+type ApiInfo struct {
+	url   string
+	qps   int
+	token chan struct{}
+}
+
+func (a *ApiInfo) Require() {
+	if a.qps > 0 {
+		a.token <- struct{}{}
+	}
+}
+func (a *ApiInfo) Release() {
+	if a.qps > 0 {
+		time.AfterFunc(time.Duration(1.0/float32(a.qps)*1000)*time.Millisecond, func() {
+			<-a.token
+		})
+	}
+}
+func (a *ApiInfo) SetQPS(qps int) {
+	a.qps = qps
+	a.token = make(chan struct{}, qps)
+}
+func (a *ApiInfo) NowLen() int {
+	return len(a.token)
+}
+func InitApiInfo(url string, qps int) *ApiInfo {
+	return &ApiInfo{
+		url:   url,
+		qps:   qps,
+		token: make(chan struct{}, qps),
+	}
+}
+
+type File struct {
+	FileName     string `json:"filename"`
+	Size         int64  `json:"size"`
+	CreateAt     string `json:"createAt"`
+	UpdateAt     string `json:"updateAt"`
+	FileId       int64  `json:"fileId"`
+	Type         int    `json:"type"`
+	Etag         string `json:"etag"`
+	S3KeyFlag    string `json:"s3KeyFlag"`
+	ParentFileId int    `json:"parentFileId"`
+	Category     int    `json:"category"`
+	Status       int    `json:"status"`
+	Trashed      int    `json:"trashed"`
+}
+
+func (f File) GetHash() utils.HashInfo {
+	return utils.NewHashInfo(utils.MD5, f.Etag)
+}
+
+func (f File) GetPath() string {
+	return ""
+}
+
+func (f File) GetSize() int64 {
+	return f.Size
+}
+
+func (f File) GetName() string {
+	return f.FileName
+}
+
+func (f File) CreateTime() time.Time {
+	parsedTime, err := time.Parse("2006-01-02 15:04:05", f.CreateAt)
+	if err != nil {
+		return time.Now()
+	}
+	return parsedTime
+}
+
+func (f File) ModTime() time.Time {
+	parsedTime, err := time.Parse("2006-01-02 15:04:05", f.UpdateAt)
+	if err != nil {
+		return time.Now()
+	}
+	return parsedTime
+}
+
+func (f File) IsDir() bool {
+	return f.Type == 1
+}
+
+func (f File) GetID() string {
+	return strconv.FormatInt(f.FileId, 10)
+}
+
+var _ model.Obj = (*File)(nil)
+
+type BaseResp struct {
+	Code     int    `json:"code"`
+	Message  string `json:"message"`
+	XTraceID string `json:"x-traceID"`
+}
+
+type AccessTokenResp struct {
+	BaseResp
+	Data struct {
+		AccessToken string `json:"accessToken"`
+		ExpiredAt   string `json:"expiredAt"`
+	} `json:"data"`
+}
+
```

Comment:
增加 Refresh Token 请求返回结构体。

```suggestion

type RefreshTokenResp struct {
	AccessToken  string `json:"access_token"`
	ExpiresIn    int    `json:"expires_in"`
	RefreshToken string `json:"refresh_token"`
	Scope        string `json:"scope"`
	TokenType    string `json:"token_type"`
}

```